CN: ampersands are not escaped in HTML output, allowing ampersand escapes in output

Here is a file that produces the &reg code which renders as ®
```
void f(void)
{
    int region = 1;
    /*@ assert(false); @*/
}
```
```
% cn verify region.c    
[1/1]: f -- fail
region.c:4:9: error: Unprovable constraint
    /*@ assert(false); @*/
        ^~~~~~~~~~~~~~ 
Constraint from region.c:4:9:
    /*@ assert(false); @*/
        ^~~~~~~~~~~~~~ 
State file: file:///var/folders/tz/g0kxkm2915xdtm6qgv8rff000000gp/T/state__region.c__f.html
```
In the state file:
```
Available Resources
Owned<signed int>(®ion)(1'i32)
...
```

--- a/backend/cn/lib/pp.ml
+++ b/backend/cn/lib/pp.ml
@@ -349,3 +349,12 @@ let document_of_yojson (json : Yojson.Safe.t) : (document, string) Result.t =
   | `String str -> Ok (PPrint.arbitrary_string str)
   | _ ->
     Error ("document_of_yojson: expected `String, found " ^ Yojson.Safe.to_string json)
+
+
+let string str =
+  let amp = Str.regexp "&" in
+  if !html_escapes then (
+    let str = Str.global_replace amp "&amp;" str in
+    string str)
+  else
+    string str

